### PR TITLE
Bug 1235300 - [GijTV] Smart system has to set the ready-state attribute to let JSMarionette know ready to test

### PR DESCRIPTION
**Pull request by Fischer-L**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0696fa6d68c011d02a7048f519a96c23ae6ac6fb

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a4f8b2cbeeeeba0ce1cd7b13deb4b5bc30c7e7e6

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c1e5e8c7f1f48ca06052663cde14f129f40f64a3

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=42b159903f3045d7d8e07334f74a377922829168

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e4fd6b35bf588635648c8b474297ab483ad9496c

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js [hunk 1]
```diff
@@ -78,9 +78,11 @@ marionette.skip('Test Context Menu Events', function() {
   });
 
   test('press enter on second menu', { 'devices': ['tv'] }, function() {
-    launchContextMenu();
-    client.switchToFrame();
+    // Long press on a link
+    var link = client.helper.waitForElement('#link');
+    actions.longPress(link, 1.5).perform();
 
+    client.switchToFrame();
```
Comment (evanxd) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Looks like we don't need this line. We'll change to System frame in the `system.waitForEvent`[1] method.
[1]: https://github.com/mozilla-b2g/gaia/blob/master/tv_apps/smart-system/test/marionette/lib/system.js#L64

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js [hunk 2]
```diff
@@ -103,9 +111,11 @@ marionette.skip('Test Context Menu Events', function() {
   });
 
   test('press esc after menu shown', { 'devices': ['tv'] }, function() {
-    launchContextMenu();
-    client.switchToFrame();
+    // Long press on a link
+    var link = client.helper.waitForElement('#link');
+    actions.longPress(link, 1.5).perform();
 
+    client.switchToFrame();
```
Comment (evanxd) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Same [1]
[1]: https://github.com/mozilla-b2g/gaia/pull/33762/files?diff=split#r49295368

File: tv_apps/smart-system/test/marionette/app_modal_dialog_test.js [hunk 3]
```diff
@@ -57,19 +54,16 @@ marionette.skip('Test Modal Dialog Events', function() {
...
   setup(function() {
-    actions = client.loader.getActions();
     system = client.loader.getAppClass('smart-system', 'system', 'tv_apps');
+    system.waitForFullyLoaded();
     appModalDialog = new AppModalDialog(client);
-  });
-
-  function launchModalDialogApp() {
     // Launch test app
+    client.switchToFrame();
```
Comment (evanxd) [inline tv_apps/smart-system/test/marionette/app_modal_dialog_test.js]: I think we don't need to switch to System frame, we still stay in System frame at this line.

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js [hunk 4]
```diff
@@ -41,34 +39,36 @@ marionette.skip('Test Context Menu Events', function() {
...
   setup(function() {
-    actions = client.loader.getActions();
     system = client.loader.getAppClass('smart-system', 'system', 'tv_apps');
-  });
-
-  function launchContextMenu() {
+    system.waitForFullyLoaded();
+    actions = client.loader.getActions();
     // Launch test app
+    client.switchToFrame();
```
Comment (evanxd) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Same[1]

[1]: https://github.com/mozilla-b2g/gaia/pull/33762/files?diff=split#r49295424

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js [hunk 5]
```diff
@@ -41,34 +39,36 @@ marionette.skip('Test Context Menu Events', function() {
...
+  test('press enter on first menu', { 'devices': ['tv'] }, function() {
     // Long press on a link
     var link = client.helper.waitForElement('#link');
     actions.longPress(link, 1.5).perform();
 
-  }
-
-  test('press enter on first menu', { 'devices': ['tv'] }, function() {
-    launchContextMenu();
     client.switchToFrame();
```
Comment (evanxd) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Same[1]
[1]: https://github.com/mozilla-b2g/gaia/pull/33762/files?diff=split#r49295368

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f7fa603768dc7d49950d1541bb205852884a7cd3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ad4e1f317ca44aef1089d00f1ef02056049cd89d

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js (hunk 1 above)
Comment (Fischer-L) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Got it , thank you

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js (hunk 2 above)
Comment (Fischer-L) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Got it , thank you

File: tv_apps/smart-system/test/marionette/app_modal_dialog_test.js (hunk 3 above)
Comment (Fischer-L) [inline tv_apps/smart-system/test/marionette/app_modal_dialog_test.js]: Got it , thank you

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js (hunk 4 above)
Comment (Fischer-L) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Got it , thank you

File: tv_apps/smart-system/test/marionette/browser_context_menu_test.js (hunk 5 above)
Comment (Fischer-L) [inline tv_apps/smart-system/test/marionette/browser_context_menu_test.js]: Got it , thank you

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=80b46a267baa03dc3061760cb8f1588deb0d3d4b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c674d289fa12507921d7a01ff689c1b9c6ba470c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=765518f39f2774d34e3300ffb218dbd52a4fb7a4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4b3f733a21bfecba5352a7ea90e1632792b8de55

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=037ebc8796038218516aea29274a62e83c7d8f39

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=90d008e1682e1fc06a6c1b6d94980b53f756d176

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2ad4c5b9e9871a26cfbfb512e4718aef444db628

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a6aa7fab2ec05f449311d08225425204a535c3c8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0b177891ef4f76c5fbc9ea74a3b3b94ba749a885

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d7f34e202d1cae74de1b1904be60e986904dfdd3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e52ce1933242183ebee5b3fbf8b40eeaeaa67780

File: tv_apps/smart-system/bower_components/smart-modal-dialog/smart-input-dialog.js [hunk 6]
```diff
@@ -177,7 +177,7 @@
     this._createButtonGroup(options);
 
     this._focusedGroupIndex = 0;
-    this._focusedGroupIndex[INPUT_GROUP_INDEX] = 0;
+    this._focusedIndex[INPUT_GROUP_INDEX] = 0;
```
Comment (luke-chang) [inline tv_apps/smart-system/bower_components/smart-modal-dialog/smart-input-dialog.js]: Remember to modify the same line in the "smart-modal-dialog" repository.

File: tv_apps/smart-system/test/marionette/lib/app_modal_dialog.js [hunk 7]
```diff
@@ -48,55 +48,107 @@ ModalDialog.Selector = Object.freeze({
 ModalDialog.prototype = {
 
   get alertDialog() {
-    return this.client.findElement(ModalDialog.Selector.alert.dialog);
+    if(!this._alertDialog) {
```
Comment (luke-chang) [inline tv_apps/smart-system/test/marionette/lib/app_modal_dialog.js]: Why do we need to cache these elements?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=aa889d8c0aaae41905f584ab62941111452c7415

File: tv_apps/smart-system/bower_components/smart-modal-dialog/smart-input-dialog.js (hunk 6 above)
Comment (Fischer-L) [inline tv_apps/smart-system/bower_components/smart-modal-dialog/smart-input-dialog.js]: Yes, has been updated, thank you

File: tv_apps/smart-system/test/marionette/app_modal_dialog_test.js [hunk 8]
```diff
@@ -96,52 +94,58 @@ marionette.skip('Test Modal Dialog Events', function() {
...
     function() {
-      launchModalDialogApp();
       client.executeAsyncScript(function(options) {
         window.wrappedJSObject.showDialog('prompt', options.prompt.message);
         marionetteScriptFinished();
       }, [options]);
       client.switchToFrame();
 
+      var promptDialog = appModalDialog.promptDialog;
       appModalDialog.waitForDialogOpened(appModalDialog.promptDialog);
```
Comment (luke-chang) [inline tv_apps/smart-system/test/marionette/app_modal_dialog_test.js]: `appModalDialog.waitForDialogOpened(promptDialog);`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=28c6a9b6c10e78174d656cf0c9fb795147f299a9

File: tv_apps/smart-system/test/marionette/app_modal_dialog_test.js (hunk 8 above)
Comment (Fischer-L) [inline tv_apps/smart-system/test/marionette/app_modal_dialog_test.js]: Updated, thanks

**Pull request merged**